File: sdk/go/examples/quick_start.go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/DeLTa-X-Tunisia/SmaRTC/sdk/go/smartc"
)

/*
 * 🚀 SmaRTC Go Quick Start
 *
 * Démontre le workflow de base :
 * 1. Login
 * 2. Créer un appel
 * 3. Lister les appels
 * 4. Terminer l'appel
 * 5. Logout
 */

func main() {
	fmt.Println("🚀 SmaRTC Go Quick Start")
	fmt.Println(string(make([]byte, 40)))

	// Initialiser le client avec logs activés
	config := &smartc.Config{
		APIBaseURL:  "http://localhost:8080",
		EnableLogs:  true,
		Timeout:     10 * time.Second,
	}
	client := smartc.NewClient(config)

	// 1. Login
	fmt.Println("\n🔐 Connexion...")
	err := client.Login("alice", "password123")
	if err != nil {
		log.Fatalf("❌ Erreur login : %v", err)
	}
	fmt.Printf("✅ Connecté en tant que : %s\n", client.CurrentUsername())

	// 2. Créer un appel
	fmt.Println("\n📞 Création d'un appel...")
	session, err := client.StartCall("Réunion Backend")
	if err != nil {
		log.Fatalf("❌ Erreur création appel : %v", err)
	}
	fmt.Println("✅ Appel créé :")
	fmt.Printf("   - Session ID : %s\n", session.SessionID)
	fmt.Printf("   - Room Name  : %s\n", session.RoomName)
	fmt.Printf("   - Host       : %s\n", session.HostUserID)

	// 3. Lister les appels disponibles
	fmt.Println("\n📋 Appels en cours...")
	calls, err := client.GetAvailableCalls()
	if err != nil {
		log.Fatalf("❌ Erreur liste appels : %v", err)
	}
	fmt.Printf("✅ %d appel(s) actif(s)\n", len(calls))
	for _, call := range calls {
		fmt.Printf("   - %s (%d participant(s))\n", call.RoomName, len(call.Participants))
	}

	// 4. Récupérer les serveurs ICE
	fmt.Println("\n🧊 Serveurs ICE...")
	iceServers, err := client.GetICEServers()
	if err != nil {
		log.Printf("⚠️ Erreur ICE : %v (utilisation Google STUN)", err)
	} else {
		fmt.Printf("✅ %d serveur(s) ICE disponible(s)\n", len(iceServers))
		for _, server := range iceServers {
			fmt.Printf("   - %v\n", server.URLs)
		}
	}

	// 5. Simuler un appel de 3 secondes
	fmt.Println("\n⏳ Appel en cours (3s)...")
	time.Sleep(3 * time.Second)

	// 6. Terminer l'appel
	fmt.Println("\n🔴 Fin de l'appel...")
	err = client.EndCall()
	if err != nil {
		log.Fatalf("❌ Erreur fin appel : %v", err)
	}
	fmt.Println("✅ Appel terminé")

	// 7. Déconnexion
	fmt.Println("\n👋 Déconnexion...")
	err = client.Logout()
	if err != nil {
		log.Fatalf("❌ Erreur logout : %v", err)
	}
	fmt.Println("✅ Session fermée")

	fmt.Println("\n" + string(make([]byte, 40)))
	fmt.Println("🎉 Terminé avec succès !")
}

File: sdk/rust/examples/quick_start.rs
use smartc::{Config, SmaRTCClient, SmaRTCError};
use std::time::Duration;

/*
 * 🦀 SmaRTC Rust Quick Start
 *
 * Démontre le workflow de base :
 * 1. Login
 * 2. Créer un appel
 * 3. Lister les appels
 * 4. Terminer l'appel
 * 5. Logout
 */

#[tokio::main]
async fn main() -> Result<(), SmaRTCError> {
    println!("🦀 SmaRTC Rust Quick Start");
    println!("{}", "=".repeat(40));

    // Initialiser le client avec logs activés
    let config = Config {
        api_base_url: "http://localhost:8080".to_string(),
        signal_server_url: "http://localhost:5001".to_string(),
        timeout: Duration::from_secs(10),
        enable_logs: true,
    };

    let mut client = SmaRTCClient::new(Some(config));

    // 1. Login
    println!("\n🔐 Connexion...");
    client.login("alice", "password123").await?;
    println!(
        "✅ Connecté en tant que : {}",
        client.current_username().unwrap()
    );

    // 2. Créer un appel
    println!("\n📞 Création d'un appel...");
    let session = client.start_call("Réunion Rust").await?;
    println!("✅ Appel créé :");
    println!("   - Session ID : {}", session.session_id);
    println!("   - Room Name  : {}", session.room_name);
    println!("   - Host       : {}", session.host_user_id);

    // 3. Lister les appels disponibles
    println!("\n📋 Appels en cours...");
    let calls = client.get_available_calls().await?;
    println!("✅ {} appel(s) actif(s)", calls.len());
    for call in &calls {
        println!(
            "   - {} ({} participant(s))",
            call.room_name,
            call.participants.len()
        );
    }

    // 4. Récupérer les serveurs ICE
    println!("\n🧊 Serveurs ICE...");
    let ice_servers = client.get_ice_servers().await?;
    println!("✅ {} serveur(s) ICE disponible(s)", ice_servers.len());
    for server in &ice_servers {
        println!("   - {:?}", server.urls);
    }

    // 5. Simuler un appel de 3 secondes
    println!("\n⏳ Appel en cours (3s)...");
    tokio::time::sleep(Duration::from_secs(3)).await;

    // 6. Terminer l'appel
    println!("\n🔴 Fin de l'appel...");
    client.end_call().await?;
    println!("✅ Appel terminé");

    // 7. Déconnexion
    println!("\n👋 Déconnexion...");
    client.logout().await?;
    println!("✅ Session fermée");

    println!("\n{}", "=".repeat(40));
    println!("🎉 Terminé avec succès !");

    Ok(())
}
